mitodl/ocw-hugo-themes
# feat: update headings

#### Pre-Flight checklist


#### What are the relevant tickets?
closes #1097 

#### What's this PR do?

Currently, if we apply Heading 1, Heading 2, and Heading 3 from OCW Studio for a course, they render as `<h3>`, `<h3>`, and `<h4>` respectively.
This PR will change this markup behaviour to HTML tags `<h3>`,` <h4>`, and `<h5>`, respectively and also apply all the other heading-related styling mentioned in [OCW Style Guide](https://projects.invisionapp.com/d/#/console/21537330/468374061/preview) for h1, h2, h3, h4, h5.

#### How should this be manually tested?
We will need to verify that that heading 1, heading 2, heading 3 are rendered to h3, h4, and h5 respectively.
We will also need to verify that all the headings style conform to the heading guidelines mentioned in the [OCW Style Guide](https://projects.invisionapp.com/d/#/console/21537330/468374061/preview)
This will need to be checked manually for www and course_v2 themes, for each value of desktop and mobile.
One way to test this would be to use inspect element and verify the values are used correctly.

**For course_v2:**
Here we have `14px=1rem`
In this case, the OCW Style Guide will have to be converted accordingly (it has 16px=1rem by default). So, eg. h1 title, mentioned as 2rem becomes: (16/14) * 2 = ~2.28rem
I have converted these values here:
https://github.com/mitodl/ocw-hugo-themes/blob/a4690c2307328ec9605c11215a74afc43be47915/base-theme/assets/css/variables.scss#L76-L121

First of all checkout this branch.
1. Clone this [course](https://github.mit.edu/ocw-content-rc/18.05-spring-2014) in your ocw-content-rc directory,
2. In your .env, `OCW_TEST_COURSE=18.05-spring-2014`
3. `yarn start course`
4. Go to this link: http://localhost:3000/pages/instructor-insights/
5. Make sure that the styling of headings correspond to the styling guidelines mentioned [here](https://projects.invisionapp.com/d/#/console/21537330/468374061/comments)
(In current code, `h4` and `h5` headings appear as `div`. This is something that will be dealt with in a new issue; so for the scope of this PR this can be ignored for now)

**Now to test that heading 1, heading 2, and heading 3 render to `h3`,` h4,` `h5`:**
1. Clone this test [course](https://github.mit.edu/ocw-content-rc/test101-spring-2023). 
2. In your .env, `OCW_TEST_COURSE=test101-spring-2023`
3. `yarn start course`
4. Go to this link: http://localhost:3000/pages/a-page-full-of-headers/
5. Check that heading 1, heading 2, and heading 3 are now `h3`, `h4`, and `h5` respectively.

**For www:**
First of all checkout this branch.
Here we have `16px=1rem`
1. `yarn start www`
2. Go to this link: http://localhost:3000/search/
3.  Verify `Explore OpenCourseWare` heading is `h1` and according to guidelines.
4. Go to this link: http://localhost:3000/
5. Verify that `Featured Courses` heading is `h2` and according to guidelines.
6. Go to this link: http://localhost:3000/newsletter/
7. Verify that the newsletter headings are now mobile responsive and correctly `h1` and `h2.` according to guidelines.

#### Screenshots (if appropriate)
Before it looked like:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/109785089/224089769-2a1e24b3-b295-4009-8204-457793e1bb60.png">

Now:
<img width="748" alt="image" src="https://user-images.githubusercontent.com/109785089/225085874-ee3613b8-06f7-4a96-939c-8dd894f5a120.png">
